src/Io/MultipartParser.php

```diff
@@ -134,14 +139,30 @@ private function parseFile($headers, $body)
         $this->request = $this->request->withUploadedFiles($this->extractPost(
             $this->request->getUploadedFiles(),
             $this->getFieldFromHeader($headers['content-disposition'], 'name'),
-            new UploadedFile(
-                Psr7\stream_for($body),
-                strlen($body),
-                UPLOAD_ERR_OK,
+            $this->parseUploadedFile($headers, $body)
+        ));
+    }
+
+    private function parseUploadedFile($headers, $body)
+    {
+        $bodyLength = strlen($body);
+        if ($this->maxFileSize !== null && $bodyLength > $this->maxFileSize) {
+            return new UploadedFile(
+                Psr7\stream_for(''),
+                0,
```

Comment:
Couldn't find anything in PHP's documentation about this (http://php.net/manual/en/reserved.variables.files.php), but it's my understanding that we should still set the file size as given by the client here so consumers can show a more descriptive error message ("tried to upload X bytes, only Y allowed")?

Reply:
SGTM :+1: